### PR TITLE
Rialto 368 use weak ptr for web audio config

### PR DESCRIPTION
**Pull request by skyproudmanp**
Summary: Use weak_ptr for passing WebAudioConfig instead of c-style pointer
Type: Cleanup
Test Plan: Passed unit tests, run on Sky stream UK box, and passed the ytcert manual webaudio test
Dependencies and Impacts: firebolt::rialto::WebAudioPlayerFactory interface has changed to accept a weak_ptr
Jira: RIALTO-368

### CODE REVIEW

File: tests/common/misc/WebAudioUtil.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2022 Sky UK
```
Comment (skywilliamsonl) [inline tests/common/misc/WebAudioUtil.cpp:5]: 2023

File: tests/common/misc/WebAudioUtil.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2022 Sky UK
```
Comment (skywilliamsonl) [inline tests/common/misc/WebAudioUtil.h:5]: 2023

**Review by skywilliamsonl — COMMENTED** [review]
https://ytlr-cert.appspot.com/latest/main.html?&test_type=manual-test&tests=47#1699287449917

**Review by skywilliamsonl — APPROVED** [review]

**Comment by skyproudmanp** [comment]
> We should also double check webaudio fullstack functionality before delivery (requires rialto-gstreamer fix too).
> 
> There is a ytcert manual webaudio test, you should be able to run it on the set top box and press arrow keys to hear beeps. https://ytlr-cert.appspot.com/latest/main.html?&test_type=manual-test&tests=47#1699287449917

Skywilliamsonl kindly ran the ytcert test yesterday and it passed

File: media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp [hunk 3]
```diff
@@ -102,14 +102,14 @@ class WebAudioPcmCapsBuilder : public WebAudioCapsBuilder
...
-                 const WebAudioConfig *config)
+                 std::weak_ptr<const WebAudioConfig> config)
     : m_context{context}, m_gstWrapper{gstWrapper}, m_glibWrapper{glibWrapper}, m_audioMimeType{audioMimeType}
 {
     RIALTO_SERVER_LOG_DEBUG("Constructing SetCaps");
 
     if (m_audioMimeType == "audio/x-raw")
     {
-        m_config.pcm = config->pcm;
+        m_config.pcm = config.lock()->pcm;
```
Comment (skywojciechowskim) [inline media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp:112]: Using lock without nullptr check may be dangerous :)

Comment (skyproudmanp) [reply media/server/gstplayer/source/tasks/webAudio/SetCaps.cpp]: @skywojciechowskim the ptr can be null therefore I added a check that throws an appropriate error before using (similar to media/client/main/source/WebAudioPlayer.cpp:92 and `media/server/main/source/WebAudioPlayerServerInternal.cpp:105)

**Review by skywojciechowskim — APPROVED** [review]

**Review by skywilliamsonl — APPROVED** [review]

**Pull request merged**